### PR TITLE
Contributing guide

### PR DESCRIPTION
**Pull request by patricoferris**
This PR adds a contributing guide primarily aimed at new developers. This is meant to be useful with respect to the outreachy proposal I am currently writing. It also add `opam depext` to the `make deps` target, on my mac the site requires `gmp, openssl, pkg-config`. 

The guide focuses on: 

 - Finding a way to contribute
 - Installation of the tools need to build and develop the site
 - General web-dev tools that might be useful
 - What our CI/Deployability check is 

This is somewhat a WIP and I'd be interested in what others have to say? :))

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ - `make deps`: We've already encountered this target, it ensures you have all of the correct dependencies to build the site.
+ - `make clean`: This will completely remove all of the build artifacts, this can be useful but also it removes the cached blog feeds so rebuilding can take a lot longer. 
+
+### Development Process
+
+This site was built quite a while ago so it doesn't have the modern features of hot-reloading and the like. Instead development is done by tweaking things and re-running `make`. Some things can be done more easily. If you want to see a CSS change quickly you can modify one in the build directory (`./ocaml.org/css`) to see it in the browser (just be sure to copy the change back to the source CSS directory!).
```
Comment (ghost) [inline CONTRIBUTING.md:53]: If you expect a few contributors, can introduce `inotifywait` or similar to almost achieve hot reload. 

For css, one can also modify the styles on the live page DOM in dev tools.

**Comment by nibble0101** [comment]
Hi @patricoferris,
A great write-up to get a beginner up and running.  I am a total beginner on matters `ocaml`. I had to google a couple of things before I was able to set up the project on my machine successfully. I  ran into `libgmp-dev` error along the way. I had to run the command described in [this](https://github.com/realworldocaml/book/issues/2992) issue. That error is probably specific to ubuntu.  After successfully  installing `ocaml` and `opam` on my computer, I remember getting the error below after  running `make deps`  or `make` (not sure at which stage exactly) at the root of the project.
```
ocamlfind
-bash: ocamlfind: command not found
```
Some of the errors above could  be due to my installation process or my system, `ubuntu 20.04 LTS`, requirements. I thought they are worth mentioning here so that you can investigate and give beginners a heads up if it is a general problem. These instructions will be great for on-boarding beginners. For the record I will be an outreachy applicant if my  initial application gets approved.

**Comment by patricoferris** [comment]
Thanks for the feedback @nibble0101! 

The `libgmp-dev` problem should actually be fixed by this PR (I've added `opam depext` to the `deps` target). 

I'm not entirely sure why you got the `ocamlfind` problem. I'm thinking it is likely you ran `make` before `make deps` and so kicked off the compilation process which [uses ocamlfind](https://github.com/ocaml/ocaml.org/blob/master/Makefile.common#L19). We should probably update the dockerfiles so that they use `make deps` so we can track that more closely. At the moment the two are not synchronised.

File: CONTRIBUTING.md (hunk 1 above)
Comment (patricoferris) [reply CONTRIBUTING.md:53]: I agree with the CSS point will add something to the dev-tools below. 

The `inotifywait` one is perhaps a little advanced for a contributing guide (it's also only Linux right?). Have you given it a go? I was under the impression the build step was... a little long for this to be feasibly useful but happy to be incorrect about that :)) ?

Comment (ghost) [reply CONTRIBUTING.md:53]: > Have you given it a go? 

I can't remember whether I have used it directly or not. I think I used build tools that wrapped inotifywait.

> little advanced for a contributing guide

Agreed.

**Comment by nayaldivya** [comment]
Hi @patricoferris, Sorry for commenting here. I've applied for Outreachy this time and looking forward to contribute to Ocaml projects. I'm a complete beginner. Any good first issues I can start with? Kindly guide me. I am really serious about contributing some quality code.  Also, tell me where to ask such questions. Thank you for your time and patience.

**Comment by patricoferris** [comment]
Hi @nayaldivya 

> Sorry for commenting here. 

That's completely fine :))

> I've applied for Outreachy this time and looking forward to contribute to Ocaml projects. 

Great 🐫 !

> Any good first issues I can start with? Kindly guide me. 

I have just tagged you in an issue which is now listed just above, it's all yours if that is something you feel comfortable tackling.

> Also, tell me where to ask such questions.

Any questions can be asked on that issue or in a PR related to that issue. All questions are important so don't hesitate to ask. I would also be interested on your feedback for this PR on adding a beginner-friendly contributing guide once you get started :)) 

Thanks for looking into contributing to the OCaml community.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+### Development Process
+
+This site was built quite a while ago so it doesn't have the modern features of hot-reloading and the like. Instead development is done by tweaking things and re-running `make`. Some things can be done more easily. If you want to see a CSS change quickly you can modify one in the build directory (`./ocaml.org/css`) to see it in the browser (just be sure to copy the change back to the source CSS directory!). 
+
+Other nice tools you can use include: 
+
+ - [Chrome Dev Tools](https://developers.google.com/web/tools/chrome-devtools) or [Firefox Dev Tools](https://developer.mozilla.org/en-US/docs/Tools): both sets of tools make it much easier to inspect HTML, CSS and Javascript in a running website. 
+ - [Google Lighthouse](https://developers.google.com/web/tools/lighthouse/), [WAVE](https://wave.webaim.org/) and [Pa11y](https://github.com/pa11y/pa11y): these are all tools for checking accessibility standards.
+
```
Comment (ghost) [inline CONTRIBUTING.md:59]: Maybe a brief section on communication conventions for PRs - should people create draft PRs? How/where should developers ask questions when working on a branch? Feel free to defer on this.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+Contributing to OCaml.org
+=========================
+
+Welcome to OCaml.org and thank you for taking the time to read our contributing guide. This guide is aimed towards first-time contributors and even people who are new to OCaml language. If you already know some concepts then feel free to skip to a section that interests you.
```
Comment (avsm) [inline CONTRIBUTING.md]:
```suggestion
Welcome to OCaml.org and thank you for taking the time to read our contributing guide. This guide is aimed towards first-time contributors and even people who are new to the OCaml language. If you already know some concepts then feel free to skip to a section that interests you. 
```

adding missing `the `

**Comment by avsm** [comment]
Fantastic first cut at the CONTRIBUTING.md; many thanks @patricoferris! Please do feel free to refine this in further PRs with @kanishka-work's comments and any other improvements you may have.

**Comment by Srinithyee** [comment]
Hi @somya104!
A fellow Outreachy applicant here. Since the community is overwhelmed with the number of applicants interested in contributing to OCaml, they suggest to create our own issues. If you find that there is something in the omcal.org website that interests you and can be fixed by you , do create a new issue. Besides, you can always throw in your doubts and questions as comments in issue #1245. That is specially meant for this purpose :)

**Comment by ghost** [comment]
People might benefit from more detailed description of the page inspector's features. Screenshots or links to tutorials explaining the page inspector's features might help

**Comment by ghost** [comment]
for accessibility, here is one cheatsheet that might be useful to link to (https://www.unr.edu/cabnr/brand/lists-and-guides/accessibility/web-accessibility-cheat-sheet). Feel free to browse around and find better cheatsheets.

---

Also, if we want to learn more about how large projects write contributing guides, people can browse the documentation at the Zulip project. Zulip is known for hosting some of the world's largest python hackathons

**Comment by Lontchi12** [comment]
Hello everyone i'm an outreachy applicant i'm having issues in setting up the project any help here?

**Pull request merged**